piccolo/columns/m2m.py

```diff
@@ -311,11 +311,8 @@ async def run(self):
         transaction, or wrapped in a new transaction.
         """
         engine = self.rows[0]._meta.db
-        if engine.transaction_exists():
-            await self._run()
```

Comment:
We don't need this check any more, because Piccolo no longer raises an exception if trying to create a transaction when one already exists (it's a no-op).